tests/test_regression/TestControllers.py

```diff
@@ -30,3 +30,48 @@ def test_regression_test_status_toggle(self):
                 self.assertEqual('False', response.json['active'])
             else:
                 self.assertEqual('True', response.json['active'])
+
+    def category_delete_if_will_abort_due_to_lack_of_permission(self):
+        """
+        This will test if it will abort on lack of permission
+        :return:
+        """
+        response = self.app.test_client().get('/regression/category/9432/delete')
+        self.assertEqual(response.status_code, 500)
+
+    def category_delete_if_will_throw_404(self):
+        """
+        Check if it will throw an error 404
+        :return:
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response_regression = c.get('/regression/test/9432/delete')
+            self.assertEqual(response_regression.status_code, 404)
+
+    def category_delete(self):
+        """
+        Check it will delete the test
+        :return:
+        """
+        # Create Valid Entry
+        from mod_regression.models import Category, RegressionTest
+
+        category = Category('Some Random Category', 'Todo')
```

Comment:
You don't need to add categories. They have been added already in base.py. This category will not have id 1.